Documentation/design-docs/native-hosting.md

```diff
@@ -306,6 +307,38 @@ Closes a host context.
 * `host_context_handle` - handle to the initialized host context to close.
 
 
+### Loading managed components
+To load managed components from native app directly (not using COM or WinRT) the hosting layer exposes a new runtime helper/delegate `hdt_load_assembly_and_get_function_pointer`. Calling the `hostfxr_get_runtime_delegate(handle, hdt_load_assembly_and_get_function_pointer, &helper)` returns a function pointer to the runtime helper with this signature:
+```C
+int load_assembly_and_get_function_pointer_fn(
+    const char_t *assembly_path,
+    const char_t *type_name,
+    const char_t *method_name,
+    const char_t *delegate_type_name,
+    void         *reserved,
+    /*out*/ void **delegate)
+```
+
+Calling this function will load the specified assembly in isolation (into its own `AssemblyLoadContext`) and it will use `AssemblyDependencyResolver` on it to provide dependency resolution. Once loaded it will find the specified type and method and return a native function pointer to that method. The method's signature can be specified via the delegate type name.
```

Comment:
Do we want to talk about reuse using `assembly_path` as a dictionary key?

Reply:
I thought about that. I added a description of this and why components should not rely on it,,,